packages/api/src/test/messaging/MessagingClient.test.ts

```diff
@@ -0,0 +1,157 @@
+  }
+  mockedConnection = {
+    close: jest.fn(),
+    createConfirmChannel: jest.fn().mockReturnValue(mockedChannel)
+  }
+
+  mockedAmqp.connect.mockReturnValue(mockedConnection as any)
+  __test__.destroyInstance()
+})
+
+test('emit', async () => {
+  const client = MessagingClient.getInstance()
+  const eventType = EventMessage.LogLogged
+
+  const expectedMessage = { pattern: eventType, data: 'the message' }
+  const expectedArgs = [eventType, '', Buffer.from(JSON.stringify(expectedMessage))]
+
+  await client.emit(eventType, expectedMessage.data)
+
+  expect(mockedChannel.publish).toHaveBeenCalledWith(...expectedArgs)
+})
+
+test('query', async () => {
+  const client = MessagingClient.getInstance()
+  const queryType = QueryMessage.Example
+
+  const expectedMessage = { pattern: queryType, id: uuidv4(), data: 'the message' }
+  const expectedArgs = [
+    queryType,
+    '',
+    Buffer.from(JSON.stringify(expectedMessage)),
+    {
+      replyTo: __test__.RANDOM_QUEUE_NAME,
+      correlationId: uuidv4()
+    }
+  ]
+
+  const promise = client.query(queryType, expectedMessage.data)
+
+  setImmediate(() => {
+    replyCallback.bind(client)({
+      properties: {
+        correlationId: uuidv4()
+      },
+      content: 'the response'
+    })
+  })
+
+  await promise
+
+  expect(mockedChannel.assertQueue).toHaveBeenCalled()
+  expect(mockedChannel.publish).toHaveBeenCalledWith(...expectedArgs)
+})
+
+test('waits for confirms before resolving emit', async () => {
+  mockedChannel.publish = jest.fn().mockReturnValue(false)
+
+  const client = MessagingClient.getInstance()
+  const eventType = EventMessage.LogLogged
+
+  const expectedMessage = { pattern: eventType, data: 'the message' }
+
+  await client.emit(eventType, expectedMessage.data)
+
+  expect(mockedChannel.publish).toHaveBeenCalled()
+  expect(mockedChannel.waitForConfirms).toHaveBeenCalled()
+})
+
+test('waits for confirms before resolving query', async () => {
+  mockedChannel.publish = jest.fn().mockReturnValue(false)
+
+  const client = MessagingClient.getInstance()
+  const queryType = QueryMessage.Example
+
+  const expectedMessage = { pattern: queryType, data: 'the message' }
+
+  const promise = client.query(queryType, expectedMessage.data)
+
+  setImmediate(() => {
+    replyCallback.bind(client)({
+      properties: {
+        correlationId: uuidv4()
+      },
+      content: 'the response'
+    })
+  })
+
+  await promise
+
+  expect(mockedChannel.publish).toHaveBeenCalled()
+  expect(mockedChannel.waitForConfirms).toHaveBeenCalled()
+})
+
+describe('error paths', () => {
+  it('throws with no connection', async () => {
+    mockedAmqp.connect.mockImplementation(() => {
+      throw new Error('connection failure test!')
+    })
+
+    const client = MessagingClient.getInstance()
+
+    await expect(client.connect(false)).rejects.toThrow()
+  })
+
+  it('quietly ignores replies for unknown correlation IDs', async () => {
```

Comment:
wouldn't this test pass whether the correlation id was found or not? could we assert that replyCallbacks.delete wasn't called to ensure the function returned earlier?

Reply:
It would, yeah. The only purpose of this test is to ensure that a known non-existant value is handled gracefully. I'll add a comment to the test to clarify.